### ISSUE TITLE
Prisma Client crashes with Node.js 16.0.0

### ISSUE THREAD

**Issue by Jolg42** [prisma/prisma#6682]
## Bug description

This error actually comes from our dependency `undici` and was patched in 3.3.5
https://github.com/nodejs/undici/issues/753

Note: it’s not yet recommended to use Node 16 because LTS start date is for the end of the year 2021-10-26   

```
 $ ts-node ./script.ts
/opt/hostedtoolcache/node/16.0.0/x64/bin/node[2518]: ../src/node_http_parser.cc:567:static void node::{anonymous}::Parser::Initialize(const v8::FunctionCallbackInfo<v8::Value>&): Assertion `args[3]->IsInt32()' failed.
 1: 0xb12b00 node::Abort() [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 2: 0xb12b7e  [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 3: 0xb2a282  [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 4: 0xd5f70b  [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 5: 0xd60bac  [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 6: 0xd61226 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
 7: 0x160c579  [/opt/hostedtoolcache/node/16.0.0/x64/bin/node]
Aborted (core dumped)
error Command failed with exit code 134.
```

## How to reproduce

Install Node 16 and use Prisma Client

## Expected behavior

No crash.

## Environment & setup

- OS: Ubuntu
- Database: any
- Node.js version: 16
- Prisma version: latest dev

**Comment by Jolg42** [comment]
Adding a test in e2e https://github.com/prisma/e2e-tests/pull/1645

**Comment by Jolg42** [comment]
undici patch merged in https://github.com/prisma/prisma/pull/6681 and released in internal `2.22.0-dev.25` will be part of next release.

**Comment by Jolg42** [comment]
Confirmed working with `2.22.0-dev.25` in e2e test.

We recommend staying on latest Node.js LTS version (Node 14).

Next Prisma version `2.22.0` will include this fix, the release is planned on May 4th.

**Comment by dickfickling** [comment]
@Jolg42 any chance we could get this fix ported back to 2.21?

**Comment by Jolg42** [comment]
Hi @dickfickling 

We do not plan to patch 2.21 with this fix at the moment since Node v16.0.0 just got out and is not recommended to use for production.

Our dev version `2.22.0-dev.25` or `dev` can be used to try things out if needed. Important to consider: we may break things and introduce unstable changes/features without warnings in the dev versions, so I would not recommend using it in production.

The next Prisma version, 2.22 will have the fix.

**Comment by DevelopmentByDavid** [comment]
Hi all, 

I think the bug still exists after upgrading prisma to `2.22.1` 

I get the error below, with almost the same environment (node --version is `v16.1.0`).  I just updated my computer yesterday.  So maybe 16.1 is newer than the fix itself.  
```
/usr/bin/node[29913]: ../src/node_http_parser.cc:567:static void node::{anonymous}::Parser::Initialize(const v8::FunctionCallbackInfo<v8::Value>&): Assertion `args[3]->IsInt32()' failed.
 1: 0x559faab35bd1 node::Abort() [/usr/bin/node]
 2: 0x559faab35c67  [/usr/bin/node]
 3: 0x559faab519b5  [/usr/bin/node]
 4: 0x559faad77c67 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [/usr/bin/node]
 5: 0x559faad78a99  [/usr/bin/node]
 6: 0x559faad7906f  [/usr/bin/node]
 7: 0x559faad79316 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/usr/bin/node]
 8: 0x559fab5c8559  [/usr/bin/node]
```

I changed to using the lts version of node and the error is gone.  It seems like that is the best fix, but thought I should still post the above information.

**Comment by janpio** [comment]
We have a successful end to end test using Node 16.1.0 with the latest `dev` version of Prisma, which will be released later today as `2.23.0`: https://github.com/prisma/e2e-tests/runs/2606119084 So we should be good here @DevelopmentByDavid. If you have a reproduction with that version later, certainly let us know though.
